src/ecalc/libraries/libecalc/common/libecalc/core/consumers/legacy_consumer/component.py

```diff
@@ -183,12 +187,38 @@ def evaluate(
 
         elif self._consumer_dto.component_type == ComponentType.PUMP:
             # Using generic consumer result as pump has no specific results currently
-            consumer_result = GenericComponentResult(
+
+            inlet_rate_time_series = TimeSeriesRate(
```

Comment:
No, I don't remember, sorry.

Reply:
@frodehk All these timeseries needs to be reindexed in order to make sure that we get the correct resulting time vector that we want, e.g. if the time vector for a pump consists of mid-year changes, but the user requests yearly resolution. It works well as long as all the changes are on the standard input time vector, but gets wrong and possibly a risk that we map incorrect value to incorrect timestep in the final resulting time vector. At least, that is what I seem to remember and understand from this.

Also, we should probably consider whether this functionality should be a part of the timeseries instead of remembering to have to deal with the vectors correctly for each and one of them...

Please correct me if I am wrong...